cmd/createcluster.go

```diff
@@ -121,6 +125,28 @@ func runCreateCluster(ctx context.Context, w io.Writer, conf clusterConfig) erro
 		return errors.Wrap(err, "mkdir")
 	}
 
+	if conf.DefFile != "" {
+		def, err := loadDefinition(ctx, conf.DefFile)
+		if err != nil {
+			return err
+		}
+		def.DefinitionHash = nil
+
+		if err = def.VerifyPartialSignatures(); err != nil {
+			return err
+		}
+		if err = def.VerifyPartialHashes(); err != nil {
+			return err
+		}
+
+		log.Warn(ctx, "Overriding cluster config with values from fetched definition since `definition-file` flag present", nil)
+		newConf, err := overrideClusterConfig(conf, def)
```

Comment:
we can't just pick a few things from the partial definition, we need to use the whole thing and only append operators. Since the creator signature needs to remain valid and this depends on the config_hash remaining the same. Please ensure this via tests

Reply:
How can the `config_hash` remain the same? Since partial definition doesn't contain any operator, but here we add operators which changes the config_hash. How do we then update the `creator_config_signature` since the original signature will get outdated? The creator needs to sign the updated definition again, is that the case?

Reply:
`config_hash` only contains operator addresses, which isn't present in this case, so the config_hash MUST remain the same. The `creator_config_signature` MUST remain valid.